packages/types/src/types.ts

```diff
@@ -55,4 +55,37 @@ export interface SlidevMarkdown {
   themeMeta?: SlidevThemeMeta
 }
 
+/**
+ * Preparser modes: listing the built-in ones but it is open to extensions.
+ * Modes starting with `:` are pseudo-modes (usually keeping the cursor in place) so that extensions can handle them.
+*/
+export type SlidevPreparserMode = (
```

Comment:
I am not so sure about the complexity here. I would prefer to have simple string transformation for the entire markdown file and then probably for each slide. I am expecting:

```ts
export interface SlidevPreparserExtension {
  name: string,
  transformRawMarkdown?(content: string): string | undefined,
  transformSlide?(no: number, content: string, frontmatter: object): string | undefined
}
```

Reply:
I understand. I need to think more about it: in particular, I'm wondering if it would require that any small transformer re-implement the parser and the serializer.

I think the "injection" ugliness would still be required or do you see another solution?
> - I ended up with a solution that makes the node module (which has the get...Roots() information) inject an "addon loader" inside the parser which invokes it during parsing, when the headmatter is closed
> - also, I'm not sure the injecting code is in the best place?